[RHACS] Release Notes for 3.70 (#45372)
openshift/openshift-docs: release_notes/370-release-notes.adoc

:_content-type: ASSEMBLY
[id="release-notes-370"]
= Red Hat Advanced Cluster Security for Kubernetes 3.70
include::modules/common-attributes.adoc[]
:context: release-notes-370

toc::[]

*Release date:* 23 May 2022

{product-title} is an enterprise-ready, Kubernetes-native container security solution that protects your vital applications across build, deploy, and runtime. It deploys in your infrastructure and integrates with your DevOps tooling and workflows to deliver better security and compliance and to enable DevOps and InfoSec teams to operationalize security.

[id="about-this-release-370"]

{product-title} ({product-title-short}) 3.70 includes feature enhancements, bug fixes, scale improvements, and other changes.

[id="new-features-370"]
== New features and enhancements

[id="verify-image-signatures-rn"]
=== Verifying image signatures against Cosign public keys

> You can use {product-title-short} to ensure the integrity of the container images in your clusters by verifying image signatures against pre-configured keys. You can also create policies to block unsigned images and images that do not have a verified signature and enforce the policy by using an admission controller to stop unauthorized deployment creation. Cosign key signature verification is supported. See _Verifying image signatures_ in the {product-title-short} Operating documentation for more information.

Reviewer: ```suggestion
You can use {product-title-short} to ensure the integrity of the container images in your clusters by verifying image signatures against preconfigured keys. You can also create policies to block unsigned images and images that do not have a verified signature and enforce the policy by using an admission controller to stop unauthorized deployment creation. Cosign key signature verification is supported. See _Verifying image signatures_ in the {product-title-short} Operating documentation for more information.
```

[id="integration-ecs-rn"]
=== Autogeneration of Amazon ECR registry integrations

> Registry integrations for Amazon Elastic Container Registry (ECR) are now auto-generated for Amazon Web Services (AWS) clusters. This feature requires that the nodes' Instance Identity and Access Management (IAM) Role has been granted access to ECR. You can turn off this feature by disabling the EC2 instance metadata service in your nodes. See _Amazon ECR integrations_ in the {product-title-short} Integrating documentation for more information.

Reviewer: ```suggestion
Registry integrations for Amazon Elastic Container Registry (ECR) are now automatically generated for Amazon Web Services (AWS) clusters. This feature requires that the nodes' Instance Identity and Access Management (IAM) Role has been granted access to ECR. You can turn off this feature by disabling the EC2 instance metadata service in your nodes. See _Amazon ECR integrations_ in the {product-title-short} Integrating documentation for more information.
```

[id="identify-missing-policies-rn"]
=== Identifying missing Kubernetes network policies

Kubernetes network policies are vital in helping to enable zero-trust networking within a cluster. They reduce the impact of network attacks by limiting the opportunity for lateral movement. By default, Kubernetes resources are not isolated. Applying network policies is a recommended best practice left to the end user.

{product-title-short} 3.70 ships with a new default policy that allows you to easily identify deployments that are not restricted by any ingress network policy and to trigger violation alerts accordingly.

- The default policy is named *Deployments should have at least one ingress Network Policy*. It is disabled by default.

- This default policy uses a new policy criterion called *Alert on missing ingress Network Policy*.

- To identify pod isolation gaps, you can clone this default policy or create a new one by using the policy criterion and enabling it on selected resources.

> [id="spring-vulnerabilities"]

Reviewer: (not a problem, but the convention on other features seems to have these end in `-rn`)

=== Identifying Spring critical vulnerabilities

> A policy to detect Spring Cloud Function RCE vulnerability (CVE-2022-22963) and Spring Framework Spring4Shell RCE vulnerability (CVE-2022-22965) has been added. It is associated with the Critical severity and is enabled by default.

Reviewer: Seems to be missing an article - "a" or "the", not sure which is appropriate.
```suggestion
A policy to detect the Spring Cloud Function RCE vulnerability (CVE-2022-22963) and Spring Framework Spring4Shell RCE vulnerability (CVE-2022-22965) has been added. It is associated with the Critical severity and is enabled by default.
```

> A policy to detect Spring Cloud Function RCE vulnerability (CVE-2022-22963) and Spring Framework Spring4Shell RCE vulnerability (CVE-2022-22965) has been added. It is associated with the Critical severity and is enabled by default.

Reviewer: "It is associated with the Critical severity" - not sure what this means

Author: Oh, yeah, this was clunky. Rewrote!

[id="allow-privilege-escalation-rn"]
=== Improved validation of pod security context

A new policy criterion has been added to validate the value of `allowPrivilegeEscalation` within the Kubernetes security context. You can use this policy criterion to provide alerts when a deployment is configured to allow a container process to gain more privileges than its parent process.

> [id="admin-credentials-link-ocp"]

Reviewer: (not a problem, but the convention on other features seems to have these end in `-rn`)

=== Finding ACS admin credentials easily in the {ocp} console

> For customers deploying {product-title-short} to {ocp} using the recommended Operator method, {product-title-short} admin credentials are now directly visible from the {ocp} console. When viewing the Central object, the *Details* tab provides a clickable link to the credentials under *Admin Password Secret Reference*. The displayed credentials are the default generated password or a previously-configured and stored custom secret. See xref:../installing/install-ocp-operator.adoc#verify-central-install-operator_install-ocp-operator[Verifying Central installation] for more information.

Reviewer: usually a capitalized item in front of the word "object" is something like an API object and gets formatted as monospaace. Not sure about this one, but thought I should mention.

Remove hyphen for "previously configured" per style guide :)

```suggestion
For customers deploying {product-title-short} to {ocp} using the recommended Operator method, {product-title-short} admin credentials are now directly visible from the {ocp} console. When viewing the `Central` object, the *Details* tab provides a clickable link to the credentials under *Admin Password Secret Reference*. The displayed credentials are the default generated password or a previously configured and stored custom secret. See xref:../installing/install-ocp-operator.adoc#verify-central-install-operator_install-ocp-operator[Verifying Central installation] for more information.
```

Author: Apparently it's not normally formatted as monospace, although when I made changes to the Install documentation recently, I formatted it as monospaced without knowing that. I'll go back and fix those so it's all consistently not formatted as monospace. (734 instances of no monospace vs. 12 monospaced. oops.)

[id="scope-increase-rn"]
=== Increased number of allowed inclusion and exclusion scopes
Previously, {product-title-short} limited the number of allowed inclusion and exclusion scopes within a scope to ten each. This restriction has been removed.

[id="notable-tech-changes-370"]
== Notable technical changes

> Vulnerability scanning and reporting for {op-system-first} nodes has been disabled until scanning improvements are made for improved accuracy and to support full host-level scanning beyond just Kubernetes components. Currently, {op-system} uses National Vulnerability Database (NVD) vulnerability data for reporting vulnerabilities for Kubernetes components from {op-system}. In the enhanced version, vulnerability reporting will be based on Red Hat published security data. (ROX-10662)

Reviewer: There's only one change here, but should it have a subheading (for the sake of consistency with a release that has multiple)?

> Vulnerability scanning and reporting for {op-system-first} nodes has been disabled until scanning improvements are made for improved accuracy and to support full host-level scanning beyond just Kubernetes components. Currently, {op-system} uses National Vulnerability Database (NVD) vulnerability data for reporting vulnerabilities for Kubernetes components from {op-system}. In the enhanced version, vulnerability reporting will be based on Red Hat published security data. (ROX-10662)

Author: Oh, good point. They aren't linkable since ROX JIRA tickets apparently aren't public.

[id="deprecation-notices-370"]
== Deprecated and removed features

Some features available in previous releases have been deprecated or removed.

> Deprecated functionality is still included in {product-title-short} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments. For the most recent list of major functionality deprecated and removed, refer to the table below. Additional information on some removed or deprecated functionality is available after the table.

Reviewer 2: Change 'Additional information on' to 'Additional information about'. (IBM Style Guide Word usage).

In the table, features are marked with the following statuses:

- GA: General Availability
- TP: Technology Preview
- DEP: Deprecated
- REM: Removed

.Deprecated and removed features tracker
[cols="3,1,1,1",options="header"]
|====
|Feature |{product-title-short} 3.68 |{product-title-short} 3.69 | {product-title-short} 3.70

|Ability to delete default policies
|DEP
|DEP
|REM

|Ability to add comments to alerts and processes
|GA
|DEP
|DEP

|Anchore, Tenable, and Docker Trusted registry integrations
|GA
|DEP
|DEP

|External authorization plug-in for scoped access control
|GA
|DEP
|DEP

|`FROM` option in the Disallowed Dockerfile line policy field
|GA
|GA
|DEP

|`RenamePolicyCategory` and `DeletePolicyCategory` API endpoints
|GA
|GA
|DEP

| `--rhacs` option for the `roxctl` helm output command
| GA
| DEP
| DEP

|Security policies without a `policyVersion`
|DEP
|DEP
|REM

|`/v1/policies` API endpoint response: *field* response body parameter.
|DEP
|DEP
|REM

|====

[id="deprecated-features-370"]
=== Deprecated features

> ** *Anchore*, *Tenable*, and *Docker Trusted Registry* integrations: The {product-title-short} scanner supersedes these integrations.

Reviewer: The other two bullets have everything before the `:` in bold. Should this match that?

> ** *Anchore*, *Tenable*, and *Docker Trusted Registry* integrations: The {product-title-short} scanner supersedes these integrations.

Author: Yep, it's Anchore :)

[id="bug-fixes-3670"]
== Bug fixes

* When configuring a JFrog Artifactory integration, the username and password fields are now optional to allow anonymous pulls. (ROX-10090)
* Validation to the web user interface for endpoint URLs in the generic webhook integration caused errors. This issue was fixed. (ROX-9902)
* The policy *OpenShift: Kubeadmin Secret Accessed* is no longer triggered if the request was from the default OpenShift `oauth-apiserver-sa` service account, because this is an expected access pattern for the OpenShift API server. (ROX-10018)
* The ability to enable notifications for multiple policies selected in the Policies list has been reinstated. To enable notifications, select one or more policies and choose Enable notification from the Bulk Actions menu. (ROX-9985)
* Read/write permission issue for vulnerability reports has been fixed. (ROX-9880)
* The ability to enable notifications for multiple policies selected in the Policies list has been reinstated. To enable notifications, select one or more policies and choose *Enable notification* from the *Bulk Actions* menu. (ROX-9985)
* Read/write permission issue for vulnerability reports has been fixed. (ROX-9880)

> * Fixed issue that caused connection problems to the {ocp} console after connecting to {product-title-short} or the inability to connect to {product-title-short} if a connection to the {ocp} console existed. Central will now respond with a 421 Misdirected Request status code to requests where the `ServerName` sent via TLS SNI does not match the `:authority` (Host) header. This feature can be turned off by setting the environment variable `ROX_ALLOW_MISDIRECTED_REQUESTS=true`. (ROX-9625)

Reviewer 2: For 'sent via TLS', the IBM Style Guide - Word usage section has guidance on whether 'via' is accepted or if it should be replaced with 'using, etc.) I think you are OK here, but PTAL

Author: Oh, good suggestion - I checked and since TLS is a transport mechanism, I think via is OK here too.
